app/controllers/v0/sessions_controller.rb

```diff
@@ -76,20 +76,20 @@ def saml_logout_callback
     def saml_callback
       saml_response = OneLogin::RubySaml::Response.new(params[:SAMLResponse], settings: saml_settings)
       @sso_service = SSOService.new(saml_response)
-      if @sso_service.persist_authentication!
+      authentication_persisted = @sso_service.persist_authentication!
+      if @sso_service.new_session && authentication_persisted
```

Comment:
totally agree... I can't see how it's happening but the stack trace points to this path (though I could totally be missing something) the additional logging in this seemingly contradictory state is to try to figure out root cause. Thinking maybe one of the error paths is returning a truthy value?

Reply:
Lets not change this, because I don't think this is what is causing the sentry issue. The better solution would be to put the guard in place on `set_sso_cookie`, adding something like `&& @session.present?`

Reply:
I also think that set_sso_cookie should not be in "after_login_actions" for semantic clarity. I like that method only getting called after the redirect has returned, but the cookie needs to be set before.